Lib/fontTools/varLib/instancer.py

```diff
@@ -196,59 +200,108 @@ def _getVarRegionScalar(location, regionAxes):
 
 
 def _scaleVarDataDeltas(varData, regionScalars):
-    # multiply all varData deltas in-place by the corresponding region scalar
-    varRegionCount = len(varData.VarRegionIndex)
-    scalars = [regionScalars[regionIndex] for regionIndex in varData.VarRegionIndex]
+    # multiply all varData deltas in-place by the corresponding region scalar, and
+    # remove regions whose scalar is 0
+    scalars = [regionScalars[ri] for ri in varData.VarRegionIndex]
     for item in varData.Item:
-        assert len(item) == varRegionCount
-        item[:] = [delta * scalar for delta, scalar in zip(item, scalars)]
+        item[:] = [
+            delta * scalar for delta, scalar in zip(item, scalars) if scalar != 0
```

Comment:
do you need != 0 or is just `if scalar` sufficient?

Reply:
`if scalar` works too but I like to be explicit when comparing numbers to 0